src/main/java/com/parasoft/report/transformer/XMLToSarif.java

```diff
@@ -96,8 +104,18 @@ private void checkInputAndOutputReportParams() {
     private void checkProjectRootPathsParam() {
         if (this.projectRootPaths != null && !this.projectRootPaths.trim().isEmpty()) {
             String[] pathsArray = this.projectRootPaths.trim().split(",");
-            // TODO: validate paths
-            this.projectRootPaths = String.join(",", Arrays.stream(pathsArray).map(String::trim).toArray(String[]::new));
+            String[] processedPaths = Arrays.stream(pathsArray).map((path) -> path.trim().replace("\\", "/")).toArray(String[]::new);
+            for (String path : processedPaths) {
+                if (!this.isAbsolutePath(path)) {
+                    throw new IllegalArgumentException(MessageFormat.format("Project root path must be an absolute path: {0}", path));
+                }
+                File rootFile = new File(path);
+                if (!rootFile.exists()) {
+                    Logger.warn(MessageFormat.format("WARN: Project root path does not exist on this machine: {0}.", path));
+                }
```

Comment:
I think we should skip this path if it is a file instead of a directory. I've tried to pass a file path to the project path parameter and results looks like:
![image](https://github.com/parasoft/parasoft-report-transformer/assets/109122853/1910906e-9755-43d8-ab85-c0289480db31)

Reply:
We assume that the user provides a directory path by default. When the given path does not exist, our code cannot determine if the provided value is a directory. The user might have generated an XML report in a non-local environment and is now converting it to a SARIF report locally. In this case, the project's source code does not exist in the local environment.